control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2500,6 +2518,35 @@ func (r *HostedControlPlaneReconciler) hostedControlPlaneInNamespace(resource cl
 	return result
 }
 
+func (r *HostedControlPlaneReconciler) etcdRestoredCondition(ctx context.Context, sts *appsv1.StatefulSet) *metav1.Condition {
+	if sts.Status.ReadyReplicas == *sts.Spec.Replicas {
+		// Check that all etcd pods have initContainers that started
+		podList := &corev1.PodList{}
+		initContainerCount := int32(0)
+		if err := r.List(ctx, podList, &client.ListOptions{
+			Namespace:     sts.Namespace,
+			LabelSelector: labels.SelectorFromValidatedSet(labels.Set{"app": "etcd"}),
+		}); err == nil {
+			for _, pod := range podList.Items {
+				for _, status := range pod.Status.InitContainerStatuses {
+					if status.Name == "etcd-init" && status.Ready {
```

Comment:
should we be also inspecting the `.ExitCode` in `.ContainerStateTerminated` here as well and capturing the `.Message` for failure scenarios in the `EtcdSnapshotRestored` condition set as false?

Reply:
Thanks @enxebre this is a good point - I'd only really considered the `true` path here, but I can definitely check the exit code so we can flag that something went wrong with the snapshot restore.

> How otherwise does a user realise if e.g they made a typo in their .restoreSnapshotURL input?

Yeah there's the typo case, but also since we rely on a pre-signed URL it could expire if it's not created right before the HostedCluster creation

The expired link problem is harder to detect, because the `curl` returns zero (so we don't exit the script), but then the snapshot restore fails, because the snapshot file contains the s3 XML response e.g `<Error><Code>AccessDenied</Code><Message>Request has expired</Message> ...`

I don't think we can inspect the URL in the code (other than perhaps to validate it's accessible) but perhaps I can look at making the script handle the case where s3 gives us XML instead of the actual snapshot more clearly :thinking:

Reply:
The latest revision checks `status.LastTerminationState.Terminated.ExitCode` and exposes the reason if a pod is found to be not ready.  

I also added a line in the script which will expose the XML error I mentioned